Client/lib/PONAPI/Client/UA.pm

```diff
@@ -0,0 +1,45 @@
+package PONAPI::Client::UA;
+
+use Moose;
+
+has type => (
+    is      => 'ro',
+    isa     => 'Str',
+    default => sub { 'Hijk' },
+);
+
+has ua_args => (
+    is      => 'ro',
+    isa     => 'ArrayRef',
+    default => sub { +[] },
+);
+
+sub BUILDARGS {
+    my $class = shift;
+    my %args = @_ == 1 ? %{ $_[0] } : @_;
+
+    if ( $args{ua_args} ) {
+        my %ua_args = @{ $args{ua_args} };
+        if ( exists $ua_args{type} ) {
+            $args{type} = $ua_args{type};
+            die "unsupported UA type: $args{type}\n"
+                unless grep { $_ eq $args{type} } qw< Hijk YAHC >;
+        }
+    }
+
+    return \%args;
+}
+
+
+sub build {
+    my $self = shift;
+
+    return (
+        $self->type eq 'Hijk' ? PONAPI::Client::UA::Hijk->new( @{ $self->ua_args } ) :
```

Comment:
Hard-coding classes? Surely, we want decoupled dependency injection.
Shouldn't the caller pass in the pre-constructed UA?